src/editor.c

```diff
@@ -5169,6 +5170,12 @@ void editor_apply_update_prefs(GeanyEditor *editor)
 	/* virtual space */
 	SSM(sci, SCI_SETVIRTUALSPACEOPTIONS, editor_prefs.show_virtual_space, 0);
 
+	/* caret Y policy */
+	caret_y_policy = CARET_EVEN;
+	if (editor_prefs.scroll_lines_around_cursor > 0)
+		caret_y_policy |= CARET_SLOP | CARET_STRICT;
```

Comment:
See http://www.scintilla.org/ScintillaDoc.html#SCI_SETYCARETPOLICY for why the options are different: if the *caret slop* is set to 0 and `CARET_STRICT` is set, it would keep the caret centered, which is not what we want.  So use Scintilla's default unless we set the *caret slop* to something non-zero, in which case these additional flags keep it working the same, but adding an *unwanted zone* near the top and bottom edges.